pandas/core/common.py

```diff
@@ -2455,6 +2455,12 @@ def is_list_like(arg):
     return (hasattr(arg, '__iter__') and
             not isinstance(arg, compat.string_and_binary_types))
 
+def is_dict_like(arg):
+    '''
+    Determin whether ``arg`` is dict-like.
+    '''
+    return hasattr(arg, '__getitem__') and hasattr(arg, 'keys')
+
```

Comment:
Not quite sure what you mean. IIUC the only dictviews are (python3) `dict.keys()`, `.items()` and `.values()`, none of which are dict like.